src/main/webapp/app/overview/course-lectures/text-unit/text-unit.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { SafeHtml } from '@angular/platform-browser';
+import { TextUnit } from 'app/entities/lecture-unit/textUnit.model';
+import { ArtemisMarkdownService } from 'app/shared/markdown.service';
+
+@Component({
+    selector: 'jhi-text-unit',
+    templateUrl: './text-unit.component.html',
+    styleUrls: ['./text-unit.component.scss'],
+})
+export class TextUnitComponent implements OnInit {
+    @Input()
+    textUnit: TextUnit;
+    isCollapsed = true;
+
+    formattedContent?: SafeHtml;
+
+    constructor(private artemisMarkdown: ArtemisMarkdownService) {}
+
+    ngOnInit(): void {
+        if (this.textUnit?.content) {
+            this.formattedContent = this.artemisMarkdown.safeHtmlForMarkdown(this.textUnit.content);
+        }
+    }
+
+    handleCollapse($event: any) {
+        $event.stopPropagation();
+        this.isCollapsed = !this.isCollapsed;
+    }
+
+    openPopup($event: any) {
+        $event.stopPropagation();
+
+        const width = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;
+        const height = window.innerHeight || document.documentElement.clientHeight || document.body.clientHeight;
+        const win = window.open('', ' ', 'toolbar=no,location=no,directories=no,status=no,menubar=no,scrollbars=yes,resizable=yes,' + `width=${width},height=${height}`);
+        win!.document.write(`<html><head><title>${this.textUnit.name}</title>`);
+        win!.document.write(
+            '<link rel="stylesheet" ' +
+                'href="https://cdnjs.cloudflare.com/ajax/libs/github-markdown-css/4.0.0/github-markdown.min.css" ' +
+                'integrity="sha512-Oy18vBnbSJkXTndr2n6lDMO5NN31UljR8e/ICzVPrGpSud4Gkckb8yUpqhKuUNoE+o9gAb4O/rAxxw1ojyUVzg==" crossorigin="anonymous" />',
```

Comment:
Please don't embed external CS like that.

Reply:
What is the preferred way to do it?

Reply:
Best way would be to serve it from /public I guess?

Reply:
Thanks for the tip @jpbernius :) 
Done in: e2d2b5836e4fe89f36926dd66c29548072587c42